internal/datastore/postgres/postgres.go

```diff
@@ -170,10 +171,8 @@ func newPostgresDatastore(
 	}
 
 	// Setup the default custom plan setting, if applicable.
-	pgConfig, err := defaultCustomPlan(parsedConfig)
-	if err != nil {
-		return nil, common.RedactAndLogSensitiveConnString(ctx, errUnableToInstantiate, err, pgURL)
-	}
+	// Setup the default query execution mode setting, if applicable.
+	pgConfig := DefaultQueryExecMode(parsedConfig)
```

Comment:
My understanding is that the `plan_cache_mode` (a server setting that tells pg how to plan) is orthogonal to the `QueryExecMode` (a client setting that tells pgx how to send queries to pg). 

Are we sure that it's okay to remove `force_custom_plan`? I would think we'd need both, but maybe I am misunderstanding the full implications of `exec`.

Reply:
Indeed it is orthogonal. If we disable prepared statements, there is no longer a reason to change the plan cache mode. We also have incentives to get rid of it, as it will make SpiceDB work out of the box with pgbouncer (requires ignoring the flag) and [RDS Proxy](https://github.com/authzed/spicedb/issues/1217#issuecomment-2191116621) (downright does not work because it does not have an option to ignore flags).

See https://github.com/authzed/spicedb/issues/1217 for more context.

Reply:
> If we disable prepared statements, there is no longer a reason to change the plan cache mode. 

I was under the impression we were already not using statement caching, and were using the mode that just uses prepared statements for type information. But it looks like that isn't correct, and I agree with you that if we just don't use prepared statements at all, there's no reason to send `force_custom_plan`